assembly/fabric8-ide-dashboard-war/pom.xml

```diff
@@ -170,4 +131,124 @@
+                        <executions>
+            <!--   3. Build dashboard as in upstream, using docker image   -->
+                            <execution>
+                                <id>build-image</id>
+                                <phase>generate-sources</phase>
+                                <goals>
+                                    <goal>run</goal>
+                                </goals>
+                                <configuration>
+                                    <target>
+                                        <!-- build user dashboard with maven -->
+                                        <exec dir="${basedir}" executable="docker" failonerror="true">
+                                            <arg value="build" />
+                                            <arg value="-t" />
+                                            <arg value="eclipse-che-dashboard" />
+                                            <arg value="--build-arg"/>
+                                            <arg value="SOURCES_DIR=./target/sources"/>
+                                            <arg value="." />
+                                        </exec>
+                                    </target>
+                                </configuration>
+                            </execution>
+            <!--   4. Get 'compiled' dashboard from dockerfile, extract to -->
+            <!--      ${project.build.directory}/dist                      -->
+            <!--      This step is identical to the upstream build         -->
+                            <execution>
+                                <id>unpack-docker-build</id>
+                                <phase>generate-sources</phase>
+                                <goals>
+                                    <goal>run</goal>
+                                </goals>
+                                <configuration>
+                                    <target>
+                                        <!-- build user dashboard with docker -->
+                                        <exec executable="bash">
+                                            <arg value="-c" />
+                                            <arg value="docker run --rm eclipse-che-dashboard | tar -C target/ -xf -" />
+                                        </exec>
+                                    </target>
+                                </configuration>
+                            </execution>
+                        </executions>
+                    </plugin>
+                </plugins>
+            </build>
+        </profile>
+        <profile>
+            <!-- Upstream native build. Activate with -Pnative -->
+            <id>native</id>
+            <build>
+                <plugins>
+                    <plugin>
+                        <artifactId>maven-antrun-plugin</artifactId>
+                        <executions>
+                            <execution>
```

Comment:
Maybe these steps should be commented as 3. and 4. as in "docker" profile? Anyway seems good to me :)

Reply:
You're probably right, I largely copied over config.

Worth noting: The native build is *slightly* different from upstream: we skip the clean step (since all artifacts are in /target already) and we defer the href update to the main build process. I'll open a new PR for documentation on this.

Reply:
PR here: https://github.com/redhat-developer/rh-che/pull/939